burn-core/src/nn/rnn/lstm.rs

```diff
@@ -139,9 +139,12 @@ impl<B: Backend> Lstm<B> {
             ),
         };
 
-        for t in 0..seq_length {
-            let indices = Tensor::arange(t..t + 1);
-            let input_t = batched_input.clone().select(1, indices).squeeze(1);
+        let [batch, _, d] = batched_input.dims();
+        for (t, input_t) in batched_input
+            .iter_dim(1)
+            .map(|e| e.reshape([batch, d]))
+            .enumerate()
+        {
```

Comment:
I'm not sure it's an improvement here, working with indices in that context in more similar with the math annotations of RNN, so it may be easier to read then when using iterators. I'm not againts providing an iterator in our tensor API, but I'm not sure this is the best place to use it.